applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -81,6 +146,8 @@ class FEASTEigensystemSolver
             "echo_level" : 0
         })");
 
+        default_params.AddMissingParameters(SettingsHelper<TScalarOut>(mParam).GetDefaultParameters());
```

Comment:
if you implement `GetDefaultParameters` i.e. add there the parameters that are specific to `double` and `complex` then the validation would already filter out a lot of invalid input and you don't have to do it manually afterwards